### PR TITLE
Fix metrics to use 2.0.2 cw logger

### PR DESCRIPTION
**Pull request by Integralist**
## Problem

Alarms aren't firing, due to dimensions associated with the alarm

## Solution

Remove the dimensions from metric calls so they will work with the latest version of Alephant CloudWatch Logger (`2.0.2`)

## Gem Release

This should be a patch release to `0.2.5`

### CODE REVIEW

File: lib/alephant/publisher/request/data_mapper_factory.rb [hunk 1]
```diff
@@ -43,10 +43,6 @@ def camalize(snake_case)
         def mapper_class_for(component_id)
           Object.const_get("#{camalize component_id}Mapper")
         end
-
-        def log_error(e, component_id, metric)
```
Comment (revett) [inline lib/alephant/publisher/request/data_mapper_factory.rb]: Is this not used anywhere?

File: lib/alephant/publisher/request/log_helper.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ module LogHelper
         include Logger
         def log_error_with_metric(e, method, message, metric = nil, metric_count = 1)
           logger.error "Publisher::Request::#{method}: '#{e.class}(#{e.message})' exception raised for: #{message}"
-          logger.metric(:name => metric, :unit => "Count", :value => metric_count) unless metric.nil?
+          logger.metric metric unless metric.nil?
```
Comment (revett) [inline lib/alephant/publisher/request/log_helper.rb]: You've removed the ability to set the `metric_count` value in the `logger.metric` call.

File: lib/alephant/publisher/request/data_mapper_factory.rb (hunk 1 above)
Comment (Integralist) [inline lib/alephant/publisher/request/data_mapper_factory.rb]: @revett I did a grep and nothing appeared to call this function

File: lib/alephant/publisher/request/log_helper.rb (hunk 2 above)
Comment (Integralist) [inline lib/alephant/publisher/request/log_helper.rb]: @revett in the latest cw logger driver code the method signature is the metric name and an optional config hash. If no hash provided then default values for `:unit` and `:value` are given (and whose values are `Count` and `1`)

File: lib/alephant/publisher/request/data_mapper_factory.rb (hunk 1 above)
Comment (revett) [inline lib/alephant/publisher/request/data_mapper_factory.rb]: :+1:

File: lib/alephant/publisher/request/log_helper.rb (hunk 2 above)
Comment (revett) [inline lib/alephant/publisher/request/log_helper.rb]: Yeah but `log_error_with_metric` allows for `metric_count` to be set (default: `1`), and then be passed to the `logger.metric` call.

That behaviour no longer exists - `metric_count` isn't used in the method.

**Comment by revett** [comment]
:cake:

**Pull request closed without merging**